.github/workflows/CI.yml

```diff
@@ -0,0 +1,84 @@
+name: CI
+# Run on master, any tag or any pull request
+on:
+  push:
+    branches: [master]
+    tags: ["*"]
+  pull_request:
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }}
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.version == 'nightly' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.3'
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
+        arch:
+          - x64
+        include:
+          # Add a 1.0 job just to make sure we still support it
+          - os: ubuntu-latest
+            version: 1.0.5
+            arch: x64
+          # Add a nightly job just to check if anything upcoming is breaking
+          # Unfortunately, this may create some noise in pull requests
+          - os: ubuntu-latest
+            version: nightly
+            arch: x64
+          # Add a 32-bit job to ensure we don't have any 64-bit specific logic
+          # Currently disabled because of issue #80
+          # - os: ubuntu-16.04
+          #   version: 1.3
+          #   arch: x86
+    steps:
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@v1
```

Comment:
I think I'm getting a bit confused with all the version numbers being in this file, would you mind explaining the differences between them?

We have:
-  `matrix: version: 1.3`, (L#15-17) [ I assume this is the Julia version? ]
- `os: version: 1.0.5`, (L#26-28)  [ I assume this is the Ubuntu version? ]

Then this line here `julia-actions/setup-julia@v1`, or is this the Julia version? I'm mainly asking because I'm curious how to add additional Julia versions to test on.

Reply:
So, the base matrix is just testing on Julia `1.3` as that's what we were doing in travis and appveyor, also what our production system runs. We're manually including a Julia `1.0.5` job in the matrix to check that we still support the LTS, but we don't want to bother running on every platform. The `uses` statements (e.g., `actions/checkout@v2`, `julia-actions/setup-julia@v1`) are specifying the versions of custom julia related actions to use in the workflow https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/finding-and-customizing-actions#using-release-management-for-your-custom-actions